src/main/java/seedu/address/logic/parser/AddPublicAddressCommandParser.java

```diff
@@ -0,0 +1,84 @@
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.addresses.Network;
+import seedu.address.model.addresses.PublicAddress;
+
+/**
+ * Parses input arguments and creates a new EditCommand object
+ */
+public class AddPublicAddressCommandParser implements Parser<AddPublicAddressCommand> {
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the EditCommand
+     * and returns an EditCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddPublicAddressCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_PUBLIC_ADDRESS_NETWORK, PREFIX_PUBLIC_ADDRESS,
+                        PREFIX_PUBLIC_ADDRESS_LABEL);
+
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddPublicAddressCommand.MESSAGE_USAGE), pe);
+        }
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_PUBLIC_ADDRESS_NETWORK, PREFIX_PUBLIC_ADDRESS,
+                PREFIX_PUBLIC_ADDRESS_LABEL)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddPublicAddressCommand.MESSAGE_USAGE));
+        }
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_PUBLIC_ADDRESS_NETWORK, PREFIX_PUBLIC_ADDRESS,
+                PREFIX_PUBLIC_ADDRESS_LABEL);
+
+        EditPersonDescriptor editPersonDescriptor = new EditPersonDescriptor();
+
+        String networkString = argMultimap.getValue(PREFIX_PUBLIC_ADDRESS_NETWORK).get();
+        String publicAddressString = argMultimap.getValue(PREFIX_PUBLIC_ADDRESS).get();
+        String publicAddressLabelString =
+                argMultimap.getValue(PREFIX_PUBLIC_ADDRESS_LABEL).get(); // okay as prefixes are confirmed to be present
+
+        Network network = ParserUtil.parseNetwork(networkString);
+        PublicAddress publicAddress = ParserUtil.parsePublicAddress(publicAddressString,
+                publicAddressLabelString, networkString);
+
+        Map<Network, Set<PublicAddress>> publicAddresses = Map.of(network, Set.of(publicAddress));
+        editPersonDescriptor.setPublicAddresses(publicAddresses);
+
+        if (!editPersonDescriptor.isAnyFieldEdited()) {
+            throw new ParseException(EditCommand.MESSAGE_NOT_EDITED);
```

Comment:
Would be good to add a test case that invokes this line i.e a unit test where no fields are edited

Reply:
Realized that this check is unnecessary because previous checks enforce that the field edited and will always be set if the execution has reached this point. Will remove in next commit.